data/pigui/modules/radar.lua

```diff
@@ -44,38 +52,99 @@ local function getColorFor(item)
+	isAutoZoom = function(self) return false end,
+}
+
+local radar = require 'PiGui.Modules.RadarWidget'()
+radar.minZoom = MIN_RADAR_SIZE
+radar.maxZoom = MAX_RADAR_SIZE
+radar.zoom = DEFAULT_RADAR_SIZE
+local radar3d = {
+	icon = icons.radar_3d,
+	auto_zoom = true,
+	size = Vector2(ui.reticuleCircleRadius * 1.8, ui.reticuleCircleRadius * 1.4),
+	getRadius = function(self) return self.size.x end,
+	zoomIn = function(self)
+		if self.auto_zoom then
+			radar.zoom = 10 ^ math.floor(math.log(radar.zoom, 10))
+		else
+			radar.zoom = math.max(radar.zoom / 10, MIN_RADAR_SIZE)
+		end
+		self.auto_zoom = false
+	end,
+	zoomOut = function(self)
+		if self.auto_zoom then
+			radar.zoom = 10 ^ math.ceil(math.log(radar.zoom, 10))
+		else
+			radar.zoom = math.min(radar.zoom * 10, MAX_RADAR_SIZE)
+		end
+		self.auto_zoom = false
+	end,
+	resetZoom = function(self)
+		radar.zoom = DEFAULT_RADAR_SIZE
+		self.auto_zoom = true
+	end,
+	getZoom = function(self) return radar.zoom end,
+	isAutoZoom = function(self) return self.auto_zoom end,
+}
+
 -- display the 2D radar
-local function display2DRadar(cntr, size)
-	local targets = ui.getTargetsNearby(current_radar_size)
-	local halfsize = size * 0.5
-	local thirdsize = size * 0.3
-	local twothirdsize = size * 0.7
+radar2d.draw = function(self, center)
+	local targets = ui.getTargetsNearby(self.zoom)
+	local halfsize = self.size * 0.5
+	local thirdsize = self.size * 0.3
+	local twothirdsize = self.size * 0.7
+	local fgColor = colors.uiPrimary
+	local bgColor = colors.uiBackground:opacity(0.54)
+	local lineThickness = 1.5
 
 	local function line(x,y)
-		-- ui.addLine(cntr + Vector2(x, y) * halfsize, cntr + Vector2(x,y) * size, colors.reticuleCircle, ui.reticuleCircleThickness)
-		ui.addLine(cntr + Vector2(x, y) * thirdsize, cntr + Vector2(x,y) * twothirdsize, colors.reticuleCircle, ui.reticuleCircleThickness)
+		-- ui.addLine(center + Vector2(x, y) * halfsize, center + Vector2(x,y) * size, colors.uiPrimaryDark, lineThickness)
+		ui.addLine(center + Vector2(x, y) * thirdsize, center + Vector2(x,y) * twothirdsize, fgColor, lineThickness)
 	end
-	ui.addCircleFilled(cntr, size, colors.lightBlueBackground, ui.circleSegments(size), 1)
-	ui.addCircle(cntr, size, colors.reticuleCircle, ui.circleSegments(size), ui.reticuleCircleThickness)
-	--	ui.addCircle(cntr, halfsize, colors.reticuleCircle, ui.circleSegments(halfsize), ui.reticuleCircleThickness)
-	ui.addCircle(cntr, thirdsize, colors.reticuleCircle, ui.circleSegments(thirdsize), ui.reticuleCircleThickness)
-	ui.addCircle(cntr, twothirdsize, colors.reticuleCircle, ui.circleSegments(twothirdsize), ui.reticuleCircleThickness)
+
+	-- radar background and border
+	ui.addCircleFilled(center, self.size, bgColor, ui.circleSegments(self.size), 1)
+	ui.addCircle(center, self.size, fgColor, ui.circleSegments(self.size), lineThickness * 2)
+
+	-- inner circles
+	--	ui.addCircle(center, halfsize, fgColor, ui.circleSegments(halfsize), lineThickness)
```

Comment:
Cleanup.

Reply:
These were in the original code and add additional lines to the 2D radar. I left them in as they may want to be enabled again in a future UI revision?

Reply:
Go ahead and leave them in then, but leave a comment inline with them mentioning the above please.

Reply:
Merged into commit 81a7203a24bd82bf69b485cd638779eac2950ead